rules/DowngradePhp70/Rector/Spaceship/DowngradeSpaceshipRector.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\Spaceship;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\BinaryOp\Smaller;
+use PhpParser\Node\Expr\BinaryOp\Spaceship;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Param;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\NodeManipulator\IfManipulator;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DowngradePhp70\Rector\Spaceship\DowngradeSpaceshipRector\DowngradeSpaceshipRectorTest
+ */
+final class DowngradeSpaceshipRector extends AbstractRector
+{
+    /**
+     * @var IfManipulator
+     */
+    private $ifManipulator;
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/QaAaWr/1
+     */
+    private const NAMESPACED_STRING_REGEX = '#\\\\#';
+
+    public function __construct(IfManipulator $ifManipulator)
+    {
+        $this->ifManipulator = $ifManipulator;
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Spaceship::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Change spaceship with check equal, and ternary to result 0, -1, 1',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+return $a <=> $b;
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+return (function ($a, $a) {
+    if ($a === $b) {
+        return 0;
+    }
+    return $a < $b ? -1 : 1;
+})($a, $b);
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param Spaceship $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $leftVariableParam = new Variable('a');
+        $rightVariableParam = new Variable('b');
```

Comment:
use `($a, $b)` as param is ok here as it just a named parameter, the actual call is depends on the value in the spaceship (left,right) that passed as "self invoked" anonymous parameter, eg:

**Before**:

```php
return getRand() <=> getRand();
```

**After**:

```php
        return (function ($a, $b) {
             if ($a === $b) {
                 return 0;
             }
             return $a < $b ? -1 : 1;
         })(getRand(), getRand());
```

Reply:
The code would not be read only by machines, but by developers too. In case of debugging working with single-letter named variables is painful, e.g. https://ocramius.github.io/blog/eliminating-visual-debt/ describes it well :) 

How could we name it better? left, right, first, second?

Reply:
updated to use `($left, $right)` https://github.com/rectorphp/rector/pull/6243/commits/6d411371f72907b1938cfbdc318adde3b27fb3a4